web/app/scripts/views/map/layers-controller.js

```diff
@@ -251,154 +262,207 @@
+                        if (set) {
+                            return TileUrlService.blackspotsUrl(set.uuid);
+                        }
+                        return undefined;
+                    })
+                ]);
+        }
+
+        function updateLayerGroups(urls) {
+            var baseRecordsUrl = urls[0];
+            var baseUtfGridUrl = urls[1];
+            var baseHeatmapUrl = urls[2];
+            var blackspotsUrl = urls[3];
+
+            updateEventLayer(baseRecordsUrl, baseUtfGridUrl);
+            updateHeatmapLayer(baseHeatmapUrl);
+            updateBlackspotLayer(blackspotsUrl);
+        }
+
+        function addBoundaryLayers() {
+            return GeographyState.getOptions()
+                .then(function(options) {
+                    var boundaryLayerOptions = angular.extend(defaultLayerOptions, {
+                        zIndex: 2
                     });
-                }
 
-                // Event record points. Use 'ALL' or record type UUID to filter layer
-                var recordsLayerOptions = angular.extend(defaultLayerOptions, {
-                    zIndex: 3
-                });
-                var recordsLayer = new L.tileLayer(
-                    ctl.getFilterQuery(baseRecordsUrl, ctl.tilekey),
-                    recordsLayerOptions);
+                    var layerPromises = options.map(function(boundary) {
+                        return TileUrlService.boundaryTilesUrl(boundary.uuid).then(
+                            function(baseBoundUrl) {
+                                var colorUrl =
+                                    (baseBoundUrl +
+                                     '?color=' +
+                                     encodeURIComponent(
+                                         boundary.color.toLowerCase()));
+                                var layer = new L.tileLayer(
+                                    colorUrl, boundaryLayerOptions);
+                                return [boundary.label, layer];
+                            }
+                        );
+                    });
 
-                // layer with heatmap of events
-                var heatmapOptions = angular.extend(defaultLayerOptions, {
-                    zIndex: 4
+                    return $q.all(layerPromises).then(function(layerLabelPairs) {
+                        ctl.boundariesLayerGroup = _.zipObject(layerLabelPairs);
+                    });
                 });
-                var heatmapLayer = new L.tileLayer(ctl.getFilterQuery(baseHeatmapUrl, ctl.tilekey),
-                    heatmapOptions);
+        }
 
-                // interactivity for record layer
-                var utfGridRecordsLayer = new L.UtfGrid(ctl.getFilterQuery(baseUtfGridUrl, ctl.tilekey), {
-                    useJsonP: false,
-                    zIndex: 5
-                });
+        function addLayerSwitcher() {
+            if (!ctl.layerSwitcher.init) {
```

Comment:
Could you accomplish the same thing by moving the check for `if (ctl.layerSwitcher)` into this function and then returning if it's not null? That way the layerSwitcher won't get initialized twice but you shouldn't have to set it to this intermediate `{init: true}` state.